libvast/vast/type.hpp

```diff
@@ -643,10 +644,15 @@ struct record_type final : recursive_type<record_type> {
   /// @returns The type at key *key* or `nullptr` if *key* doesn't resolve.
   const type* at(std::string_view key) const;
 
-  /// Retrieves the type at a given offset.
+  /// Retrieves the field at a given offset.
+  /// @param o The offset to resolve.
+  /// @returns The field at offset *o* or `nullptr` if *o* doesn't resolve.
+  const record_field* at(const offset& o) const;
+
+  /// Returns the field at the given offset with the full name starting from the
+  /// root node.
```

Comment:
The documentation mixes internals with user knowledge. Can we make it a bit clearer? What is a "root node"?